File: app/src/main/java/ru/aradxxx/ciceronetabssample/di/ActivityBindingModule.java
package ru.aradxxx.ciceronetabssample.di;

import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ru.aradxxx.ciceronetabssample.MainActivity;

@Module
public interface ActivityBindingModule {
    @ContributesAndroidInjector(modules = {
            AppActivityModule.class,
            FragmentBindingModule.class
    })
    MainActivity bindAppActivity();

    @Module()
    abstract class AppActivityModule {
        @Binds
        public abstract Context bindContext(@NonNull Application application);
    }
}

File: app/src/main/java/com/github/aradxxx/ciceroneflowssample/di/ActivityBindingModule.kt
package com.github.aradxxx.ciceroneflowssample.di

import android.app.Application
import android.content.Context
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import com.github.aradxxx.ciceroneflowssample.MainActivity

@Module
interface ActivityBindingModule {
    @ContributesAndroidInjector(
        modules = [
            AppActivityModule::class,
            FragmentBindingModule::class
        ]
    )
    fun bindAppActivity(): MainActivity

    @Module
    abstract class AppActivityModule {
        @Binds
        abstract fun bindContext(application: Application): Context
    }
}
